client/post-editor/controller.js

```diff
@@ -253,4 +253,14 @@ export default {
 		page.redirect( redirectWithParams );
 		return false;
 	},
+
+	gutenberg: async function( context, next ) {
+		const siteId = await getSelectedSiteId( context.store.getState() );
```

Comment:
this isn't an async function or something that returns a promise. Therefore, no need for the `await`.